### PR TITLE
Fixes for the latest app version

### PR DESCRIPTION
**Pull request by eu-siemann**
Verified locally. CI is failing due to the lack of disk space.

Signed-off-by: Eugene Smirnov <evgenii.smirnov@here.com>

### CODE REVIEW

File: conf/layer.conf [hunk 1]
```diff
@@ -9,3 +9,4 @@ BBFILE_COLLECTIONS += "libaktualizr-demo"
 BBFILE_PATTERN_libaktualizr-demo = "^${LAYERDIR}/"
 BBFILE_PRIORITY_libaktualizr-demo = "7"
 
+LAYERSERIES_COMPAT_libaktualizr-demo = "warrior"
```
Comment (pattivacek) [inline conf/layer.conf:12]: Should this also include zeus and dunfell as well, if it is compatible with them? Otherwise this seems like it might belong in a warrior branch.

File: recipes-sota/libaktualizr-demo/libaktualizr-demo_git.bb [hunk 2]
```diff
@@ -36,7 +37,10 @@ do_install_append () {
 
 FILES_${PN} += " \
   ${bindir}/libaktualizr-demo-app \
+  ${libdir}/libaktualizr.so \
+  ${libdir}/libaktualizr_secondary.so \
   ${libdir}/sota/sota.toml \
   ${libdir}/sota/conf.d \
   ${localstatedir}/sota/sota_provisioning_credentials.zip \
   "
+FILES_${PN}-dev = ""
```
Comment (pattivacek) [inline recipes-sota/libaktualizr-demo/libaktualizr-demo_git.bb:46]: Why is this necessary?

File: conf/layer.conf (hunk 1 above)
Comment (eu-siemann) [reply conf/layer.conf:12]: It is probably compatible, but we've never tested it. And we have only one branch here.

File: recipes-sota/libaktualizr-demo/libaktualizr-demo_git.bb (hunk 2 above)
Comment (eu-siemann) [reply recipes-sota/libaktualizr-demo/libaktualizr-demo_git.bb:46]: Otherwise, all shared libraries get into the PN-dev package and it leads to a QA error "PN rdepends on PN-dev" or something like that. We have it in meta-updater as well.

File: conf/layer.conf (hunk 1 above)
Comment (pattivacek) [reply conf/layer.conf:12]: I think something like this should be documented or made more obvious. To me this is unexpected. It's fine for the moment if it solves immediate problems. I think it's reasonable to expect that it should also be compatible with newer branches, though.

**Review by pattivacek — APPROVED** [review]
Is this CI problem is out of our control? I don't mind merging this now if we need to, but it's obviously not ideal.

**Comment by eu-siemann** [comment]
We need to migrate from the shared runner to fix the CI. Then we can add other yocto branches to the CI and LAYERSERIES_COMPAT. But I'm not sure when I'll have time to do that.

**Comment by eu-siemann** [comment]
I agree that it is not ideal, but I'll still merge it now since it at least fixes the completely broken layer.
